fuel/datasets/hdf5.py

```diff
@@ -300,20 +306,12 @@ def open(self):
         return None if self.load_in_memory else self._out_of_memory_open()
 
     def _out_of_memory_open(self):
-        file_id = self._get_file_id()
-        state = h5py.File(name=file_id, mode="r", driver=self.driver)
-        self.ref_counts[state.id] += 1
-        return state
+        self.get_file_handle()
+        return self.path
 
-    def close(self, state):
+    def close(self):
```

Comment:
This method is part of the interface defined in `Dataset` (see its definition [here](https://github.com/bartvm/fuel/blob/master/fuel/datasets/base.py#L130-L132)), and as such we should keep the `state` argument.